### DISCUSSION TITLE
Input on move hub

### DISCUSSION THREAD

**Discussion by phemmer** [pybricks/support#1085] [Q&A]
According to the [Hub to PC Communication](https://pybricks.com/projects/tutorials/wireless/hub-to-device/pc-communication/) tutorial, `input()` should be usable on the BOOST Move Hub:

> This program works on all hubs except for the BOOST Move Hub, since it does not support the `usys` or `uselect` module. You can still make it work on that hub by using the `input()` and `print()` functions instead.

However this does not appear to be the case. Even with flashing the 3.3.0b5 firmware, I get a `NameError` trying to use `input()`. Is this an error in the tutorial? Or is this an accidental omission from the firmware?  
I know the move hub is extremely limited, and that's why features are missing, and historically `input()` has been one of those missing features. But I wasn't sure if recent optimizations had changed that, and maybe that's why the tutorial was advising it.

And if `input()` is not supported, are there any other mechanisms for sending any sort of input from PC to the hub, no matter how limited? It looks like `MoveHub.ble.observe()` is functional. Is there any way to send to that from PC? I assume so, but not sure how `observe()` and `broadcast()` are implemented in bluetooth.

**Comment by dlech** [comment]
Thanks for pointing out this oversight. Unfortunately, the move hub is completely full right now so I'm not sure if we will be able to enable it or not.

I think doing broadcast/observe on Windows and Linux is theoretically possible, but certainly not trivial. And it is not technically possible on Mac. I have added a document describing the protocol at https://github.com/pybricks/technical-info/blob/master/pybricks-ble-broadcast-observe.md.
